spec/rspec/runner_spec.rb

```diff
@@ -114,37 +115,40 @@
     end
   end
 
-  describe '#run' do
-    subject(:runner) { described_class.new({}, RSpec.configuration, world) }
-    let(:world) { instance_double('RSpec::Core::World') }
-    let(:options) { ConfigurationOptions.new('test', 'test2') }
-
-    before do
-      allow(subject).to receive(:persist_example_statuses).and_return false
-      allow(world).to receive(:ordered_example_groups).and_return(%w[a b])
-    end
+  describe '#setup' do
+    subject { runner.setup(STDOUT, STDOUT) }
+    let!(:runner) { described_class.new(options, configuration, world) }
+    let(:options) { instance_double('RSpec::Core::ConfigurationOptions').as_null_object }
+    let(:world) { instance_double('RSpec::Core::World', filtered_examples: []).as_null_object }
+    let(:configuration) { instance_double('RSpec::Core::Configuration').as_null_object }
 
     context 'without examples_limit set' do
       it 'runs with world ordered example groups' do
-        expect(runner).to receive(:run_specs).with(%w[a b]).and_return(true)
-        runner.run(STDOUT, STDOUT)
+        expect(runner).not_to receive(:reconfigure_to_limit)
+        subject
       end
     end
 
     context 'with examples_limit set' do
       before do
-        ENV['CRYSTALBALL_CONFIG'] = 'spec/fixtures/crystalball.yml'
+        ENV['CRYSTALBALL_EXAMPLES_LIMIT'] = '1'
       end
+      after { ENV.delete('CRYSTALBALL_EXAMPLES_LIMIT') }
 
-      after { ENV.delete('CRYSTALBALL_CONFIG') }
-
-      it 'returns whatever ExamplesPruner returns' do
-        pruner = double(world_groups: %w[a b], pruned_groups: ['pruned_groups'])
-        allow(Crystalball::RSpec::ExamplesPruner).to receive(:new).with(world, to: 1).and_return pruner
+      it 'reconfigures RSpec env with new set from ExamplesPruner' do
+        pruner = double(pruned_set: ['pruned_set'])
```

Comment:
use `let` for this
and you can place both `allow`s into `before` block

Reply:
`let` doesn't make any sense since it is not something that should be reused.